packages/platform-sdk/test/integrationTests/product-projection/product-projection.test.ts

```diff
@@ -0,0 +1,315 @@
+    expect(
+      productProjectionResponse.body.masterVariant.attributes
+        .filter((attribute) => attribute.name === 'test-set-text')
+        .at(0).value
+    ).toContain('foo')
+    expect(
+      productProjectionResponse.body.masterVariant.attributes
+        .filter((attribute) => attribute.name === 'test-set-number')
+        .at(0).value
+    ).toContain(11.0)
+    expect(
+      productProjectionResponse.body.masterVariant.attributes
+        .filter((attribute) => attribute.name === 'test-set-integer')
+        .at(0).value
+    ).toContain(11)
+    expect(
+      productProjectionResponse.body.masterVariant.attributes
+        .filter((attribute) => attribute.name === 'test-enum')
+        .at(0).value
+    ).toEqual({ key: 'test', label: 'test' })
+
+    await deleteProduct(product)
+    await deleteProductType(productType)
+    await deleteTaxCategory(taxCategory)
+    await deleteCategory(category)
+  })
+
+  it('should query a product by product projection', async () => {
+    const category = await createCategory()
+    const taxCategory = await createTaxCategory()
+    const productType = await createProductType(productTypeDraftForProduct)
+    const productDraft = await createProductDraft(
+      category,
+      taxCategory,
+      productType
+    )
+    const product = await createProduct(productDraft)
+    const productProjectionQueryResponse = await apiRoot
+      .productProjections()
+      .get({
+        queryArgs: {
+          where: 'id=' + '"' + product.body.id + '"',
+          staged: true,
+        },
+      })
+      .execute()
+    expect(productProjectionQueryResponse).not.toBe(null)
+    expect(productProjectionQueryResponse.body.results.at(0).id).toEqual(
+      product.body.id
+    )
+
+    await deleteProduct(product)
+    await deleteProductType(productType)
+    await deleteTaxCategory(taxCategory)
+    await deleteCategory(category)
+  })
+
+  it('should search a product by product projection', async () => {
+    const category = await createCategory()
+    const taxCategory = await createTaxCategory()
+    const productType = await createProductType(productTypeDraftForProduct)
+    const productDraft = await createProductDraft(
+      category,
+      taxCategory,
+      productType
+    )
+    const product = await createProduct(productDraft)
+    const productProjectionSearchResponse = await apiRoot
+      .productProjections()
+      .search()
+      .get({
+        queryArgs: {
+          facet: 'categories.id',
+          staged: true,
+        },
+      })
+      .execute()
+    expect(productProjectionSearchResponse).not.toBe(null)
+    expect(productProjectionSearchResponse.body.facets).not.toBe(null)
+
+    await deleteProduct(product)
+    await deleteProductType(productType)
+    await deleteTaxCategory(taxCategory)
+    await deleteCategory(category)
+  })
+
+  // it('should get a product by SKU using query predicates', async () => {
+  //   const category = await createCategory()
```

Comment:
Why do we have this commented code?

Reply:
you are right :D, I already removed them in my locale I will clean them up in the next push ;)